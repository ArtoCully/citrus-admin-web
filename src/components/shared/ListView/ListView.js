// @flow
import React, { PureComponent } from 'react'
import moment from 'moment'
import { Link } from 'react-router'
import LinkReplace from 'src/components/shared/LinkReplace/LinkReplace'
import img from 'src/styles/images.css'
import css from './ListView.style.css'

type Props = {
  items: Array<any>,
  location: Object,
  showDelete?: boolean,
  deleteItemName?: string,
}

class ListView extends PureComponent {

  props: Props

  render (): React$Element<any> {

    const {
      location,
      showDelete,
      deleteItemName,
    } = this.props

    const items = this.props.items || []

    const list = items.map((item, index) => {

      let optionsRightDate = null
      let optionsRightDelete = null

      if (showDelete && deleteItemName) {

        optionsRightDate = (
          <div className={css.announcement}>
            {item.startDate && item.endDate
              ? `${
                moment(item.startDate).format('MM/DD/YYYY')
              } - ${moment(item.endDate).format('MM/DD/YYYY')}`
              : <span />
            }
          </div>
        )

        optionsRightDelete = (
          <LinkReplace
            className={`${css.deleteItem} ${img.icnTrash}`}
            to={{
              pathname: location.pathname,
              state: {
                modal: `delete-${deleteItemName}`,
                [deleteItemName]: item,
              },
            }}
          />
        )

      }


      return (
        <div key={index} className={css.item}>
          <Link
            className={css.detailsLink}
            to={`${location.pathname}/${item.id || item.deviceId}`}
          >
            <div className={`${css.leftSide} ${css.flex1}`}>
              <span className={css.name}>
                {item.name}
              </span>
              <span className={css.description}>
                {item.description || item.productLine}
              </span>
            </div>
            <div
              className={`${css.seeMoreLink} ${css.flex1}`}
            >
                Click to see more
            </div>
            {optionsRightDate}
          </Link>
          {optionsRightDelete}
        </div>
      )

    })


    return (
      <div className={css.list}>
        {list}
      </div>
    )

  }

}


export default ListView
