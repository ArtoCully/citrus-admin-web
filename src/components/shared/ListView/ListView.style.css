.list {
  margin-top: $s3;
  margin-bottom: $s3;
}
.item {
  display: flex;
  border: 1px solid $colorGray;
  border-radius: $borderRadius;
  margin: $s2 0;
  @mixin font-size 13;
  justify-content: space-between;
  align-items: center;
}
.item:hover {
  background-color: $colorGrayLight;
  color: $colorDark;

  .seeMoreLink {
    opacity: 1;
  }
}
.detailsLink {
  color: $colorDark;
  padding: $s3;
  flex: 96;
  display: flex;
  justify-content: space-between;
  align-items: center;
}
.detailsLink:hover {
  color: $colorDark;
}
.leftSide {
  display: flex;
  flex-direction: column;
}
.name {
  font-weight: bold;
  @mixin font-size 15;
}
.description {
  width: 450px;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}
.seeMoreLink {
  opacity: 0;
  color: $colorTheme;
  @mixin font-size 13;
  font-weight: 900;
  transition: opacity .5s ease-in-out;
}
.announcement {
  display: flex;
  justify-content: space-between;
}
.flex1 {
  flex-grow: 1;
}
.noBorder {
  border: 0;
}
.deleteItem {
  z-index: 100;
  flex: 4;
}
