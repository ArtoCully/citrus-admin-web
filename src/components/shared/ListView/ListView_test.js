// @flow
/* eslint-disable max-len */

// Enzyme docs:
// http://airbnb.io/enzyme/docs/api/index.html

import React from 'react'
import { shallow } from 'enzyme'
import ListView from './ListView'


const mockProps = {
  items: [],
  location: {},
}


it('<ListView> renders the ListView content', () => {

  const wrapper = shallow(
    <ListView
      {...mockProps}
    />
  )

  expect(wrapper).toMatchSnapshot()

})
