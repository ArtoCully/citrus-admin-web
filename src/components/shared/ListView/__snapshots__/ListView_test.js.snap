// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListView> renders the ListView content 1`] = `
<div
  className="list"
/>
`;
